.github/maintainers_guide.md

```diff
@@ -111,63 +111,101 @@ If you want to test the package locally you can.
 It is possible to deploy this project on <https://test.pypi.org/> in order to try out packaging and deploy related changes
```

Comment:
I don't have very much context on **test.pypi.org** or when we might want to do a development deployment. Adding a quick blurb on what this is and when it's done might be nice, but not a blocker!

Reply:
good point I've added a blurb for this 💯

Reply:
TIL the greatness of test pypi